packages/headless/src/app/engine.ts

```diff
@@ -190,6 +191,16 @@ export function buildEngine<
     );
   }
 
+  if (
+    shouldWarnAboutMismatchBetweenOrganizationIDAndOrganizationEndpoints(
+      options
+    )
+  ) {
+    engine.logger.warn(
+      `There is a mismatch between the \`organizationId\` option (${options.configuration.organizationId}) and the organization configured on the \`organizationEndpoints\` option (${options.configuration.organizationEndpoints?.platform}). This could lead to issues that are complex to troubleshoot. Please make sure both values are equivalent.`
```

Comment:
```suggestion
      `There is a mismatch between the \`organizationId\` option (${options.configuration.organizationId}) and the organization configured in the \`organizationEndpoints\` option (${options.configuration.organizationEndpoints?.platform}). This could lead to issues that are complex to troubleshoot. Please make sure both values match.`
```